prowler/providers/gcp/gcp_provider.py

```diff
@@ -316,9 +324,16 @@ def test_connection(
                     client_id, client_secret, refresh_token
                 )
 
-            session, _ = GcpProvider.setup_session(
+            session, project_id = GcpProvider.setup_session(
                 credentials_file, service_account, gcp_credentials
             )
+
+            if provider_id and project_id != provider_id:
```

Comment:
```suggestion
            if provider_id and project_id != provider_id and provider_id not in GcpProvider.get_projects(credentials=credentials):
```

but my question is why do we need to get all the projects if the setup session returns the project id.

Reply:
The setup_session returns the default project_id, could happen that the given one doesn't match with the default